### PR TITLE
feature(TTW-7104): Individual character Log Header

### PR DESCRIPTION
**Pull request by danny-janse**
As described in #7104 the character logs isnt rendering properly. With these changes this page is rendering as it should again.
In addition the armory background images have been broken for certain characters so i touched up the way these are handled also.

There are a few test cases for this, not including screenshots as this would clutter the MR and make it unreadable.

**Guild logs**
https://wowanalyzer.com/guild/US/Tichondrius/incarnate/

**Hidden Character**
* Logs: https://wowanalyzer.com/character/US/Tichondrius/Grobbulo/
* Character tab: **Unavailable**

**Regular working character**
* Logs: https://wowanalyzer.com/character/US/area-52/Sunnyflight/
* Character tab: https://wowanalyzer.com/report/FtwyZB68kzK7bLaA/85-Heroic+Queen+Ansurek+-+Kill+(6:27)/Sunnyflight/standard/character

** Not working character**
* Logs: https://wowanalyzer.com/character/US/Tichondrius/Talgek/
* Character tab: https://wowanalyzer.com/report/FtwyZB68kzK7bLaA/93-Heroic+Queen+Ansurek+-+Kill+(7:17)/Talgek/standard/character

### CODE REVIEW

File: src/interface/CharacterParses.tsx [hunk 1]
```diff
@@ -397,7 +406,16 @@ class CharacterParses extends Component<CharacterParsesProps, CharacterParsesSta
           });
           return;
         }
-        // hidden isn't a value on the parse response
+
+        // WCL responds with {hidden:true} when the logs are hidden.
+        if (rawParses?.hidden) {
```
Comment (danny-janse) [inline src/interface/CharacterParses.tsx]: Typecheck fails on `Property 'hidden' does not exist on type 'WCLParsesResponse'.`

Im not sure how to properly work around this, any suggestions?
Depending on the response `WCLParsesResponse` should be an `WCLParse[]` or a type of `{ hidden: boolean }`

File: src/game/CLASSES.ts [hunk 2]
```diff
@@ -30,6 +30,22 @@ export enum CLASSES {
   WARRIOR,
 }
 
+export const CLASS_NAMES: { [classId: number]: { name: string } } = {
```
Comment (emallson) [inline src/game/CLASSES.ts:33]: TIL that we didn't have class ids anywhere

**Review by emallson — APPROVED** [review]
the character picture logic seems fine. i updated the typing of the wcl parses response to explicitly represent (and handle) the hidden response

File: src/game/CLASSES.ts (hunk 2 above)
Comment (danny-janse) [reply src/game/CLASSES.ts:33]: I was quited perplexed by this, apparently we convert the classes from the string/spec instead.

**Pull request merged**